packages/ui/src/hooks/use-social-connector.ts

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { signInWithSocial, signInToSoical } from '@/apis/social';
+import { generateRandomString } from '@/utils';
+
+import useApi from './use-api';
+
+const storageKeyPrefix = 'social_auth_state';
+const webPlatformPrefix = 'web';
+const mobilePlatformPrefix = 'mobile';
+
+const isMobileWebview = () => {
+  // TODO: read from native sdk embedded params
+  return true;
+};
+
+const useSocial = () => {
+  const { result: signInWithSocialResult, run: asyncSignInWithSocial } = useApi(signInWithSocial);
+  const { result: signInToSoicalResult, run: asyncSignInToSoical } = useApi(signInToSoical);
+
+  const { search } = useLocation();
+
+  const validateState = useCallback((state: string, connectorId: string) => {
+    if (state.startsWith(mobilePlatformPrefix)) {
+      return true; // Not able to validate the state source from the native call stack
+    }
+
+    const stateStorage = sessionStorage.getItem(`${storageKeyPrefix}:${connectorId}`);
+
+    return stateStorage === state;
+  }, []);
+
+  const socialCallbackHandler = useCallback(
+    (connector?: string) => {
+      const uriParameters = new URLSearchParams(search);
+      const state = uriParameters.get('state');
+      const code = uriParameters.get('code');
+
+      if (!state || !code || !connector) {
+        // TODO: error message
+        return;
+      }
+
+      if (!validateState(state, connector)) {
+        // TODO: error message
+        return;
+      }
+
+      void asyncSignInToSoical({ connectorId: connector, state, code, redirectUri: 'TODO' });
+    },
+    [asyncSignInToSoical, search, validateState]
+  );
+
+  const signInWithSocialHandler = useCallback(
+    async (connectorId: string) => {
+      const state = `${
+        isMobileWebview() ? mobilePlatformPrefix : webPlatformPrefix
```

Comment:
We don't need a `state` in the native social flow in this step.
The UI gets the `state` from the native and passes the `state` to the later process. ( get access token)

Reply:
Ignore me. the `state` is needed for `/api/session/sign-in/social`